### ISSUE TITLE
Allow user-defined IP instead of resolving our own

### ISSUE THREAD

**Issue by thatguyatgithub** [jp-bennett/Fwknop2#60]
Thanks a lot for Fwknop2, I really love it and use it tons of time.

I encountered myself on many situations in which outgoing connections on HTTP(s) get resolved using a different outgoing proxy or public IP, instead of the ones used by any non HTTP(s) traffic. 

On the same fashion, I encountered other situations in which I'm requesting from my mobile phone IP's a knocking to myself on other network that had outgoing non-http(s) traffic blocked.

To sum up, there appears to exists several situation in which you would be needing to craft a fwknop knocking operation for another public IP than the one you are requesting it from, In those cases, being able to NOT resolving and passing it over an IP by hand would be quite useful.
